test/common/network/dns_impl_test.cc

```diff
@@ -929,6 +944,33 @@ TEST_P(DnsImplAresFlagsForTcpTest, TcpLookupsEnabled) {
   ares_destroy_options(&opts);
 }
 
+class DnsImplAresFlagsForNoDefaultSearchDomainTest : public DnsImplTest {
+protected:
+  bool tcpOnly() const override { return false; }
+  void updateDnsResolverOptions() override {
+    dns_resolver_options_.mutable_no_default_search_domain()->set_value(true);
+  }
+};
+
+// Parameterize the DNS test server socket address.
+INSTANTIATE_TEST_SUITE_P(IpVersions, DnsImplAresFlagsForNoDefaultSearchDomainTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+// Validate that c_ares flag `ARES_FLAG_NOSEARCH` is set when boolean property
+// `no_default_search_domain` is enabled.
+TEST_P(DnsImplAresFlagsForNoDefaultSearchDomainTest, NoDefaultSearchDomainSet) {
+  server_->addCName("root.cnam.domain", "result.cname.domain");
```

Comment:
Is `cnam` misspelt?

Reply:
I kept the original test case as-is. It probably is misspelt. I just changed it to cname and tests work without any problem.